lib/CL/devices/common.c

```diff
@@ -1446,6 +1455,18 @@ static const cl_image_format supported_image_formats[] = {
 };
 #endif
 
+/** Initializes device info defaults.
+ *
+ * It setups the most generic defaults applicable to most devices,
+ * which the device drivers can overwrite with more specific values.
+ * pocl_cpu_init_common() should be called instead for CPU-devices.
+ * Some of the parameters such as llvm_target_triplet and llvm_cpu
+ * can be set also beforehand, in which case the default host values
+ * are not used and they are used to initialize the rest of the data.
+ *
```

Comment:
Seems unnecessarily complicated....

The CMake variables like OCL_KERNEL_TARGET, OCL_KERNEL_TARGET_CPU and LLC_TRIPLE are *always* related to the host CPU. Is there any other other device where they makes sense ? So perhaps it would be better 1) move the setup of target specific variables (llvm_cpu, llvm_target_triplet etc) into the driver-specific init function (`pocl_cpu_init_common` etc) 2) keep only the kernellib setup in the `pocl_init_default_device_infos` since that can be shared.